src/components/select-project/select-project.controller.ts

```diff
@@ -83,13 +83,18 @@ export class SelectProjectController implements angular.IController {
   private canIAddToProject(): boolean {
     let canIAddToProject: boolean = true;
 
-    // TODO AuthorizationService.canIAddToProject assumes that the ProjectsService.get has already completed performed
-    // but currently this happens after project change events fire and it happens in the various dialog controllers, not
-    // as part of this component.  For now always return true until we can resolve this.
-    //
-    // if (!this.isNewProject()) {
-    //   canIAddToProject = this.AuthorizationService.canIAddToProject(_.get(this.ctrl.selectedProject, 'metadata.name'));
-    // }
+    var projectName = _.get(this.ctrl.selectedProject, 'metadata.name');
+
+    if (!this.isNewProject()) {
+      if (_.isUndefined(this.AuthorizationService.getRulesForProject(projectName))) {
+        this.AuthorizationService.getProjectRules(projectName).then( () => {
+          canIAddToProject = this.AuthorizationService.canIAddToProject(projectName);
+          return this.ctrl.forms.selectProjectForm.selectProject.$setValidity('cannotAddToProject', canIAddToProject);
+        });
+      } else {
+        canIAddToProject = this.AuthorizationService.canIAddToProject(projectName);
+      }
+    }
```

Comment:
I believe this can just be:

```ts
if (!this.isNewProject()) {
  this.AuthorizationService.getProjectRules(projectName).then( () => {
    canIAddToProject = this.AuthorizationService.canIAddToProject(projectName);
    this.ctrl.forms.selectProjectForm.selectProject.$setValidity('cannotAddToProject', canIAddToProject);
  });
}
```

It will resolve immediately without requesting the rules again if they're already cached.

Reply:
Ok, should I always call `AuthorizationService.getProjectRules(projectName, true)' to always update the cache?

Reply:
> Ok, should I always call AuthorizationService.getProjectRules(projectName, true) to always update the cache?

No, I wouldn't. I don't see a reason not to use the cached results. It doesn't change often, and the user can refresh the page if needed. I believe it's a fairly expensive request.

Reply:
Yes, that worked.